lib/matplotlib/axes/_axes.py

```diff
@@ -5409,7 +5409,10 @@ def get_interp_point(idx):
                             np.column_stack([ind[where], dep2[where]])])
         if ind_dir == "y":
             pts = pts[:, ::-1]
-        self.update_datalim(pts, updatex=True, updatey=True)
+
+        if "transform" not in kwargs:
+            self.update_datalim(pts, updatex=True, updatey=True)
```

Comment:
This should probably check the transform branches, similar to how lines do:
https://github.com/matplotlib/matplotlib/blob/690884facc1e80b40640649f2a64ceab67afb42e/lib/matplotlib/axes/_base.py#L2354

Reply:
That may be a bit more complete, yeah... for reference I based the original suggestion on axline:

https://github.com/matplotlib/matplotlib/blob/b86ebbafe4673583345d0a01a6ea205af34c58dc/lib/matplotlib/axes/_axes.py#L910-L913

Reply:
@QuLogic I'm not not yet very well-versed with transforms - wondering if you could provide some guidance as to how this would work?

I will continue to look into this as well.

Reply:
See #25324 for a similar application of the transform branches for `update_datalim` on `vlines`/`hlines`. (Though I don't know for sure that you also need to handle `rectilinear` specifically as is done in that PR; just the transform bit might be enough.)

Reply:
@QuLogic @ksunden I updated the PR to check transform branches - can you review this to make sure everything looks alright?